webknossos/webknossos/geometry/nd_bounding_box.py

```diff
@@ -0,0 +1,375 @@
+        assert (
+            match is not None
+        ), f"Could not extract bounding box from {checkpoint_name}"
+        bbox_tuple = tuple(int(value) for value in match.group().split("_"))
+        return cls.from_tuple6(cast(Tuple[int, int, int, int, int, int], bbox_tuple))
+
+    @classmethod
+    def from_csv(cls, csv_bbox: str) -> "NDBoundingBox":
+        bbox_tuple = tuple(int(x) for x in csv_bbox.split(","))
+        return cls.from_tuple6(cast(Tuple[int, int, int, int, int, int], bbox_tuple))
+
+    @classmethod
+    def from_auto(
+        cls, obj: Union["NDBoundingBox", str, Dict, List, Tuple]
+    ) -> "NDBoundingBox":
+        raise NotImplementedError
+
+    @classmethod
+    def group_boxes_with_aligned_mag(
+        cls, bounding_boxes: Iterable["NDBoundingBox"], aligning_mag: Mag
+    ) -> Dict["NDBoundingBox", List["NDBoundingBox"]]:
+        """
+        Groups the given BoundingBox instances by aligning each
+        bbox to the given mag and using that as the key.
+        For example, bounding boxes of size 256**3 could be grouped
+        into the corresponding 1024**3 chunks to which they belong
+        by using aligning_mag = Mag(1024).
+        """
+
+        chunks_with_bboxes = defaultdict(list)
+        for bbox in bounding_boxes:
+            chunk_key = bbox.align_with_mag(aligning_mag, ceil=True)
+            chunks_with_bboxes[chunk_key].append(bbox)
+
+        return chunks_with_bboxes
+
+
+    def to_wkw_dict(self) -> dict:
+        ( 
+            width,
+            height,
+            depth,
+        ) = self.size.to_list()
+
+        return {
+            "topLeft": self.topleft.to_list(),
+            "width": width,
+            "height": height,
+            "depth": depth,
+        }
+
+    def to_config_dict(self) -> dict:
+        return {"topleft": self.topleft.to_list(), "size": self.size.to_list()}
+
+    def to_checkpoint_name(self) -> str:
+        return f"{'_'.join(str(element) for element in self.topleft)}_{'_'.join(str(element) for element in self.size)}"
+
+    def to_csv(self) -> str:
+        return ",".join(map(str, self.to_tuple6()))
+
+    def __repr__(self) -> str:
+        return f"NDBoundingBox(topleft={self.topleft}, size={self.size}, axes={self.axes})"
+
+    def __str__(self) -> str:
+        return self.__repr__()
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, NDBoundingBox):
+            return self.topleft == other.topleft and self.size == other.size
+        else:
+            raise NotImplementedError()
+
+    def padded_with_margins(
+        self, margins_left: VecNIntLike, margins_right: Optional[VecNIntLike] = None
+    ) -> None:
+        raise NotImplementedError()
+
+    def intersected_with(
+        self, other: "NDBoundingBox", dont_assert: bool = False
+    ) -> "NDBoundingBox":
+        """If dont_assert is set to False, this method may return empty bounding boxes (size == (0, 0, 0))"""
+
```

Comment:
These functions should probably get assertions to check that they have the same axes.

Reply:
Yes sounds good. I think it would be useful to have a fixed order for axis names to make comparison easier. Maybe (x,y,z,<other_axes_in_alphabetical_order>)